csrc/id_model/id_model.cpp

```diff
@@ -925,17 +946,87 @@ Expr* findMatchingExpr(
       continue;
     }
 
+    // For the final loop promotion map, we want to find
+    // promotions within the same loop groups. Note that that's
+    // guaranteed when a new domain is replayed instead of reusing an
+    // existing domain.
+    if (require_loop_mapped_promotion) {
```

Comment:
Is there any reason why we don't just do the following?
```C++
    if (true) {
```

Reply:
Hmm... I understand this would avoid more replay. But wouldn't you need to pay back most of these "saved" replays in step 4 anyway?

Reply:
I am feeling that this "optimization" is just like you borrowed some money in step 2, and pay them back in step 4. Not a strong ask because the cost of updating all docs/tests can be high, but I would prefer `if (true) {` here, because this simplifies our mental model about loop promotion: if you print the diff of the IEL promotion map of each step, the diff between step 3 and step 4 can be noisy due to the change in `require_loop_mapped_promotion `, removing this "optimization" could make it clearer to understand what step 4 does.

Reply:
Yeah, I agree with you. I'll create a cleanup issue for now.

Reply:
#2027